### PR TITLE
Release v0.14.2

### PR DESCRIPTION
**Pull request by fdosani**
Release v0.14.2
----------------

### New features
- `SnowflakeCompare` support is now available!

### Changes
- Fix comparison report when one column is all NAs (#343)
- Adding `SnowflakeCompare` (Snowflake/Snowpark compare (#343)
- Snowflake compare annotation fix, add docs (#344)

### CODE REVIEW

File: docs/source/developer_instructions.rst [hunk 1]
```diff
@@ -43,6 +43,24 @@ Run ``python -m pytest`` to run all unittests defined in the subfolder
 `pytest-runner <https://pypi.python.org/pypi/pytest-runner>`_.
 
 
+Snowflake testing
```
Comment (ak-gupta) [inline docs/source/developer_instructions.rst:46]: Thank you for updating the developer docs -- we aren't always consistent with this so it's very helpful.

File: pyproject.toml [hunk 2]
```diff
@@ -56,16 +56,19 @@ python-tag = "py3"
 [project.optional-dependencies]
 duckdb = ["fugue[duckdb]"]
 spark = ["pyspark[connect]>=3.1.1; python_version < \"3.11\"", "pyspark[connect]>=3.4; python_version >= \"3.11\""]
+snowflake = ["snowflake-connector-python", "snowflake-snowpark-python"]
```
Comment (ak-gupta) [inline pyproject.toml:59]: [NON-BLOCKING]: how does `snowflake-snowpark-python` affect the dependency tree? Does it add a bunch of packages to the `snowflake` extra?

**Review by ak-gupta — APPROVED** [review]
I did a quick skim and everything looks good 🚀

File: pyproject.toml (hunk 2 above)
Comment (fdosani) [reply pyproject.toml:59]: I think it most depends on the snowflake-connector-python if you look here: https://github.com/snowflakedb/snowpark-python/blob/main/setup.py#L16

**Review by gladysteh99 — APPROVED** [review]
LGTM!

**Pull request merged**